File: example/src/main/kotlin/com/example/_03_behavioral_patterns/_17_mediator/java/_02_after/CleaningService.java
package com.example._03_behavioral_patterns._17_mediator.java._02_after;

public class CleaningService {

    private final FrontDesk frontDesk = new FrontDesk();

    public void getTowers(Integer guestId, int numberOfTowers) {
        String roomNumber = this.frontDesk.getRoomNumberFor(guestId);
        System.out.println("provide " + numberOfTowers + " to " + roomNumber);
    }
}

File: example/src/main/kotlin/com/example/_03_behavioral_patterns/_17_mediator/kt/_02_after/CleaningService.kt
package com.example._03_behavioral_patterns._17_mediator.kt._02_after

class CleaningService(
    private val frontDesk: FrontDesk = FrontDesk()
) {
    fun getTowers(guestId: Int, numberOfTowers: Int) {
        val roomNumber = frontDesk.getRoomNumberFor(guestId)
        println("provide $numberOfTowers to $roomNumber")
    }
}
